File: Java/src/com/spun/projects/test/PhoneNumberTest.java
package com.spun.projects.test;

import junit.framework.TestCase;

public class PhoneNumberTest extends TestCase {
	public void test1() {
		PhoneNumber ph = new PhoneNumber("+1 858 775 2869");
		assertEquals(ph.getStrippedNumber(), "+18587752869");
		assertEquals(ph.getValueAsNorthAmerican(), "(858)775-2869");
		assertEquals(ph.isNorthAmericanNumber(), true);
		assertEquals(ph.getValueAsInternational(), "+1.858.775.2869");
	}

	public void test2() {
		PhoneNumber ph = new PhoneNumber("1 858 775 3456");
		assertEquals(ph.getStrippedNumber(), "+18587753456");
		assertEquals(ph.getValueAsNorthAmerican(), "(858)775-3456");
		assertEquals(ph.isNorthAmericanNumber(), true);
		assertEquals(ph.getValueAsInternational(), "+1.858.775.3456");
	}

	public void test3() {
		PhoneNumber ph = new PhoneNumber("+1(858)775-2868");
		assertEquals(ph.getStrippedNumber(), "+18587752868");
		assertEquals(ph.getValueAsNorthAmerican(), "(858)775-2868");
		assertEquals(ph.isNorthAmericanNumber(), true);
		assertEquals(ph.getValueAsInternational(), "+1.858.775.2868");
	}

	public void test4() {
		PhoneNumber ph = new PhoneNumber("+1(858)775-2868x123");
		assertEquals(ph.getStrippedNumber(), "+18587752868x123");
		assertEquals(ph.getValueAsNorthAmerican(), "(858)775-2868x123");
		assertEquals(ph.isNorthAmericanNumber(), true);
		assertEquals(ph.getValueAsInternational(), "+1.858.775.2868x123");
	}

	public void test5() {
		PhoneNumber ph = new PhoneNumber("+598.123.4567x858\"");
		assertEquals(ph.getStrippedNumber(), "+18587752868");
		assertEquals(ph.getValueAsNorthAmerican(), "+5981234567x858");
		assertEquals(ph.isNorthAmericanNumber(), false);
		assertEquals(ph.getValueAsInternational(), "+598.123.456.7x858");
	}
}

File: CSharp/Interview/PhoneNumberTest.cs
using FluentAssertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Interview

{
    [TestClass]
    public partial class PhoneNumberTests
    {
        [TestMethod]
        public void Test1()
        {
            PhoneNumber ph = new PhoneNumber("+1(858)775-2868");

            ph.GetOriginalText().Should().Be("+1(858)775-2868");
            ph.GetStrippedNumber().Should().Be("+18587752868");
            ph.GetValueAsNorthAmerican().Should().Be("(858)775-2868");
            ph.GetValueAsInternational().Should().Be("+1.858.775.2868");
        }

        [TestMethod]
        public void Test2()
        {
            PhoneNumber ph = new PhoneNumber("+1(858)775-2868x123");

            ph.GetOriginalText().Should().Be("+1(858)775-2868x123");
            ph.GetStrippedNumber().Should().Be("+18587752868x123");
            ph.GetValueAsNorthAmerican().Should().Be("(858)775-2868x123");
            ph.GetValueAsInternational().Should().Be("+1(858)775-2868x123");
        }

        [TestMethod]
        public void Test3()
        {
            PhoneNumber ph = new PhoneNumber("+598.123.4567x858");

            ph.GetOriginalText().Should().Be("+598.123.4567x858");
            ph.GetStrippedNumber().Should().Be("+5981234567x858");
            ph.GetValueAsNorthAmerican().Should().Be(null);
            ph.GetValueAsInternational().Should().Be("+598.123.456.7x858");
        }

        [TestMethod]
        public void Test4()
        {
            PhoneNumber ph = new PhoneNumber("+27 1234 5678 ext 4");

            ph.GetOriginalText().Should().Be("+27 1234 5678 ext 4");
            ph.GetStrippedNumber().Should().Be("+2712345678x4");
            ph.GetValueAsNorthAmerican().Should().Be(null);
            ph.GetValueAsInternational().Should().Be("+27 1234 5678 ext 4");
        }

        [TestMethod]
        public void Test5()
        {
            PhoneNumber ph = new PhoneNumber("858-775-2868");

            ph.GetOriginalText().Should().Be("858-775-2868");
            ph.GetStrippedNumber().Should().Be("+18587752868");
            ph.GetValueAsNorthAmerican().Should().Be("(858)775-2868");
            ph.GetValueAsInternational().Should().Be("+1.858.775.2868");
        }
    }
}
